homeassistant/components/rpi_gpio.py

```diff
@@ -48,6 +48,20 @@ def setup_input(port, pull_mode):
                GPIO.PUD_DOWN if pull_mode == 'DOWN' else GPIO.PUD_UP)
 
 
+def run_pwm(hass, port, frequency, duty_cycle, duration):
```

Comment:
I suggest adding a callback as a keyword argument that defaults to `None`. If there's a callback it should be called in the end of the `stop` callback.

In this case you want to pass the `inactivate` function as a callback. Then you don't need to schedule it separately with a tracker of the same time. This is better since I think you want to make sure that the GPIO is stopped before the process is reversed.